bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,291 @@
+
+; Transaction Input List
+: [Concept] The enumeration of all Transaction Inputs of a transaction
+
+; Transaction Output
+: [Component] Creates a UTXO by specifying an amount and an output script
+
+; Transaction Output List
+: [Concept] The enumeration of all Transaction Outputs of a transaction
+
+; txid
+: [Component] Part of an Outpoint that identifies the transaction that created a UTXO. Result of hashing a transaction without its witness structure.
+
+; Version
+: [Component] Part of the transaction header that specifies the transaction version.  Serializes as the four-byte <code>nVersion</code> integer field.
+
+; Witness Item
+: [Component] An element of a witness stack. Witness Items are: Witness Script (P2WSH only), Script Arguments, Leaf Script (P2TR scriptpath only), Control Block (P2TR scriptpath only), Annex (P2TR only). BIP341 also refers to Witness Items as “Witness Elements” or “Witness Stack Elements”.
+
+; Witness Program
+: [Concept] A script template that forwards input validation to the Witness Stack. Witness Programs are a type of Forwarding Script. Witness Programs appear in the Output Script for native segwit outputs and in the Input Script for wrapped segwit inputs.
+
+; Witness Structure
+: [Concept] The part of the serialized transaction that contains the witness stacks for each input
+
+; Witness Script
+: [Component] A Witness Item that provides the Condition Script used to spend P2WSH outputs. Appears in the Witness Stack.
+
+; Witness Stack
+: [Component] The pre-built stack of witness items that is executed in validation to satisfy an input's condition script. If a transaction has at least one segwit input, each input has exactly one witness stack.  Witness stacks are not scripts, but may include witness items that are scripts. While there is a witness stack for each input, the witness stack is not part of the input. Strictly speaking, a witness stack can contain witness items that will not be added to the stack, such as the Control Block. The Witness Stack may contain Condition Scripts, specifically it will contain a Witness Script for P2WSH inputs, and a Leaf Script for a P2TR script path spend.
+
+; Witness Version
+: [Component] The version of the witness program in native segwit outputs
+
+; wtxid
+: [Concept] Witness Transaction Identifier. Result of hashing a complete transaction including its witness structure. Used in transaction propagation and to construct the Witness Commitment.
+
+===Usage in context of UTXO creation and spending===
+
+{| class="wikitable"
+! Output Type !! Output Script !! Output Script content !! Input Script !! Witness Stack
+|-
+| Bare (e.g. P2PK) || Bare Output Script || Various (e.g. <nowiki><PK> OP_CHECKSIG)</nowiki> || Various (e.g. <nowiki><SIG></nowiki>) || rowspan="2" | <nowiki>[ ]</nowiki>
+|-
```

Comment:
Would it make sense to list P2MS below P2PK?

Reply:
Bare Outputs include P2PK, P2PKH, P2MS, and any other arbitrary scripts that are fully defined in the _scriptPubKey_ directly. I previously had more listed, but it bloated the table. I’ll add more explanation as a footnote or similar.